components/patient-access-rules/api/src/test/java/org/phenotips/data/permissions/internal/DefaultCollaboratorTest.java

```diff
@@ -36,19 +48,36 @@
  */
 public class DefaultCollaboratorTest
 {
+    @Mock
+    private ComponentManager cm;
+
+    @Mock
+    private Provider<ComponentManager> mockProvider;
+
+    @Mock
+    private UsersAndGroups usersAndGroups;
+
     /** The user used as a collaborator. */
     private static final DocumentReference COLLABORATOR = new DocumentReference("xwiki", "XWiki", "hmccoy");
 
     private static final AccessLevel access = mock(AccessLevel.class);
 
-    private static final PatientAccessHelper helper = mock(PatientAccessHelper.class);
+    @Before
+    public void setupUsersAndGroups() throws ComponentLookupException
+    {
+        MockitoAnnotations.initMocks(this);
+        Utils.setComponentManager(this.cm);
+        ReflectionUtils.setFieldValue(new ComponentManagerRegistry(), "cmProvider", this.mockProvider);
```

Comment:
ComponentManagerRegistry keeps its provider in the field cmProvider. This command injects a mock provider into it, so that when anyone asks for a UserAndGroups object it will return the local mock UserAndGroups.